src/router/api/auth/signup.router.ts

```diff
@@ -0,0 +1,42 @@
+import { FastifyPluginCallback } from 'fastify';
+
+import app from '@src/app';
+import User from '@src/entity/user.entity';
+
+type TSignupParam = {
+  id: string;
+  nickname: string;
+  password: string;
+};
+
+const signupRoute: FastifyPluginCallback = (fastify, opts, done) => {
+  fastify.post('/signup', async (req, res) => {
+    const { id: userid, nickname, password } = <TSignupParam>req.body;
+
+    try {
+      const user = await User.findOne({ userid, password });
+
+      if (user) {
+        return res.status(401).send({ message: 'User-id already exists' });
+      }
+
+      const registerUser = new User();
+      registerUser.userid = userid;
+      registerUser.nickname = nickname;
+      registerUser.password = password;
```

Comment:
**Important**

Please encrypt password with library `bcrypt` for security issue